File: api/src/test/java/com/sourceforgery/swedbank/INTTest.java
package com.sourceforgery.swedbank;

import java.io.IOException;
import java.util.List;

import com.sourceforgery.swedbank.ECardClient.Account;
import com.sourceforgery.swedbank.ECardClient.ECardAPI;

import okhttp3.logging.HttpLoggingInterceptor.Level;

public class INTTest {

    public void testLoginAndGetCards() throws Exception {
        ECardClient.setDebugLevel(Level.HEADERS);

        List<Account> accounts = ECardClient.login(System.getenv("PERSONNR"));
        ECardAPI api = accounts.get(1).selectIssuer();
        RealCard realCard = api.getCards().get(0);
        List<ActiveECard> activeECards = api.getActiveECards(realCard, 0);
        System.err.println(activeECards);
        api.getPastTransactions(realCard, 0);
        System.err.println(activeECards);
    }
}

File: api/src/test/java/com/sourceforgery/swedbank/INTTest.kt
package com.sourceforgery.swedbank

import okhttp3.logging.HttpLoggingInterceptor.Level
import org.junit.Test

class INTTest {

    @Throws(Exception::class)
    @Test
    fun testLoginAndGetCards() {
        debugLevel = Level.BODY

        val eCardClient = ECardClient(System.getenv("PERSONNR"))
        eCardClient.loginWithoutPoll()
        val accounts = eCardClient.pollAndGetAccounts()
        val api = accounts[1].selectIssuer()
        val activeECards = api.getActiveECards(0)
        System.err.println(activeECards)
        api.getPastTransactions(0)
        System.err.println(activeECards)
    }
}
